internal/cluster/cluster_test.go

```diff
@@ -3172,6 +3179,9 @@ var _ = Describe("Permanently delete clusters", func() {
 		Expect(db.First(&common.Cluster{}, "id = ?", c2.ID).RowsAffected).Should(Equal(int64(0)))
 		Expect(db.Unscoped().First(&common.Cluster{}, "id = ?", c2.ID).RowsAffected).Should(Equal(int64(1)))
 
+		// ListClusterManifestsInternal(ctx, manifests.V2ListClusterManifestsParams{ClusterID: *c.Cluster.ID})
+		manifestsAPI.EXPECT().ListClusterManifestsInternal(gomock.Any(), manifests.V2ListClusterManifestsParams{ClusterID: *c1.Cluster.ID}).AnyTimes()
+		manifestsAPI.EXPECT().ListClusterManifestsInternal(gomock.Any(), manifests.V2ListClusterManifestsParams{ClusterID: *c2.Cluster.ID}).AnyTimes()
 		mockS3Api.EXPECT().DeleteObject(gomock.Any(), c1.ID.String()).Return(false, nil).Times(1)
```

Comment:
Because these functions are listing user manifests since the change.
Each file from the directory is then checked against the list of known user manifests as follows

If the file is a manifest and is not a user manifest then it is a system manifest.
In the case of legacy clusters (ones from before the "manifest metadata" change) the files are always considered to be "user manifests"